osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/domain/services/DeviceResponseService.java

```diff
@@ -0,0 +1,20 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.adapter.protocol.iec60870.domain.services;
+
+import org.opensmartgridplatform.adapter.protocol.iec60870.domain.exceptions.AsduHandlerException;
+import org.opensmartgridplatform.adapter.protocol.iec60870.domain.valueobjects.ResponseMetadata;
+import org.opensmartgridplatform.dto.da.measurements.MeasurementReportDto;
+
+@FunctionalInterface
+public interface DeviceResponseService {
+
+    abstract void process(MeasurementReportDto measurementReportDto, ResponseMetadata responseMetadata)
+            throws AsduHandlerException;
```

Comment:
Looked convenient to reuse the exception as these services are only used from within the ASDU handlers. 
Refactored code to log errors instead of throwing exceptions...